getting-started/testing-setup.md

```diff
@@ -0,0 +1,220 @@
+# Testing Setup
+
+Some of the tests we present in the guide examples utilize non-standard requirements that can be appended to your local rails helper. These additions will help provide an error-free learning experience! Let's start with our Gemfile to make sure we have the required dependencies:
+
+{% code title="Gemfile" %}
+```ruby
+group :development, :test do
+  # Preferred testing suite which we use in our Solidus guides
+  gem 'rspec-rails'
+  # Setup 'factories' or prebuilt objects to expedite testing
+  gem 'factory_bot_rails' '~> 4.8'
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+end
+
+#...
+
+group :test do
+# Adds support for Capybara system testing and selenium driver
+  gem 'capybara', '>= 3.26'
+  # Easy installation and use of web drivers to run system tests with browsers
+  gem 'webdrivers'
+  # Scrubs database clean between tests
+  gem 'database_cleaner-active_record'
+```
+{% endcode %}
+
+Let's go over the installation of these gems and why we are suggesting them!
+## RSpec
+RSpec is a highly versatile testing framework that we utilize for Solidus and in our guide examples. To install it for your local project, simply Update your bundle and install RSpec base files with the following commands in your shell:
+
+```bash
+bundle install
+rails g rspec:install
+```
+For more information on RSpec and associated methods, please see the [official documentation](https://relishapp.com/rspec/rspec-rails/docs).
+
+## Solidus Factories and authentication helpers
+
+Whether you are following along this learning guide and coding along with the examples or developing on top of the Solidus platform, you might find yourself interested in utilizing some of the pre-built factories or authentication helpers. In order use the pre-built factories, we will have to edit a few files starting with your Gemfile:
+
+{% hint style="warning" %}
+We advise to use the same version used by Solidus to ensure you don't get errors from differences in how FactoryBot builds/creates associations, but if your app was based on a more recent version you might prefer to stick to that one instead.
+{% endhint %}
+
+Next, we will need to navigate to `config/application.rb` and customize it to look like the following:
+
+{% code title="config/application.rb" %}
+```ruby
+module AmazingStore
+  class Application < Rails::Application
+
+    # … everything else
+
+    # Don't initialize this if factory_bot_rails is not part of the current bundle group.
+    if defined?(FactoryBotRails)
+      initializer after: "factory_bot.set_factory_paths" do
+        require 'spree/testing_support'
```

Comment:
I think the file you required is actually [just requiring](https://github.com/solidusio/solidus/blob/25daae38b6c87931c26506a0948b37f47e00156d/core/lib/spree/testing_support.rb#L7-L10) the stuff we have in [`spree/testing_support/factory_bot`](https://github.com/solidusio/solidus/blob/master/core/lib/spree/testing_support/factory_bot.rb).  It may be better to load that file instead, cc'ing @elia for confirmation since he knows this part better. 

Out of curiosity @cpfergus1: where did you take this code? We may want to update that source as well.

Reply:
I got it from Elia's discussion thread [#4014](https://github.com/solidusio/solidus/discussions/4014)